cli/command/image/build.go

```diff
@@ -247,6 +253,66 @@ func runBuild(dockerCli *command.DockerCli, options buildOptions) error {
 		buildCtx = replaceDockerfileTarWrapper(ctx, buildCtx, relDockerfile, translator, &resolvedTags)
 	}
 
+	// write ctx tar to disk, to make it readable for Image Reference extraction
+	originCtx := buildCtx
+	defer originCtx.Close()
+	readbackTar, err := ioutil.TempFile("", "docker-ctx")
```

Comment:
Why is it necessary to write the tar to disk? IIRC the translation for notary wraps the tar stream with a reader rewrites the Dockerfile on the fly. I would think that just extracting information from the Dockerfile is even simpler.